advanced/Scripts/version.sh

```diff
@@ -8,179 +8,93 @@
 # This file is copyright under the latest version of the EUPL.
 # Please see LICENSE file for your rights under this license.
 
-# Variables
-DEFAULT="-1"
-COREGITDIR="/etc/.pihole/"
-WEBGITDIR="/var/www/html/admin/"
-
 # Source the setupvars config file
 # shellcheck disable=SC1091
-source /etc/pihole/setupVars.conf
+. /etc/pihole/setupVars.conf
 
-getLocalVersion() {
-    # FTL requires a different method
-    if [[ "$1" == "FTL" ]]; then
-        pihole-FTL version
-        return 0
-    fi
+# Sourece the versions file poupulated by updatechecker.sh
```

Comment:
```suggestion
# Source the versions file poupulated by updatechecker.sh
```

Reply:
Why do we have codespell at all?

Reply:
@PromoFaux

Reply:
I guess souerce is not in its misspellings dictionary.

We can remove if if doesn't work

Reply:
Maybe. I got the feeling it's only detecting issues when it likes to do some work, otherwise it just says LGTM. We had some misspellings in the past that weren't detected for a long time and suddenly it's complaining.

However, we should keep it - better to correct us from time to time than never.

Reply:
> I guess souerce is not in its misspellings dictionary.

I know, but this is not the expected behavior.
That's weird. We expect it to find all errors.

It found the error here (`versons` => `versions`):
https://github.com/pi-hole/pi-hole/blob/79c0b446e194c4d7f2be249a9e250be987e7601b/advanced/Scripts/version.sh#L22

But it didn't find `Sourece`.

Reply:
I looked through all of the `codespell` dictionaries at https://github.com/codespell-project/codespell/tree/master/codespell_lib/data but none contains `Sourece`.

__

I know, but this is not the expected behavior.
That's weird. We expect it to find all errors.

Actually it is. `Codespell` only finds errors that are listed in their dictionaries. If an misspelling is not part of it, it won't pop up.

Reply:
I know how it works and I know its not easy to get every spell error, but I think it works differently than the general "expected behavior".

The "expected" behavior from a spell checker action is to find all spell errors, not to find just a small set of errors.

This action is helpful, but there are some "holes" to be fixed.

Reply:
Sure. I'm not sure why they did choose this path. Maybe because users use all kind of strange function names and variables. If the would catch everything that is not in a usual dictionary they would get way too many false-positives. Their approach produces almost non, but misses real spelling mistakes in exchange.